### PR TITLE
[Feature][Enterprise WeChat Alert]add a alert plugin of the enterprise wechat's robot

### PR DESCRIPTION
**Pull request by jon-qj**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-ewechat/pom.xml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>dolphinscheduler-alert-plugins</artifactId>
+        <groupId>org.apache.dolphinscheduler</groupId>
+        <version>2.0.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>dolphinscheduler-alert-ewechat</artifactId>
+    <packaging>jar</packaging>
```
Comment (kezhenxu94) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-ewechat/pom.xml:28]:
```suggestion
```

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-ewechat/pom.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+        </dependency>
+    </dependencies>
+
+
```
Comment (kezhenxu94) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-ewechat/pom.xml]:
```suggestion
```

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-ewechat/src/main/java/org/apache/dolphinscheduler/plugin/alert/ewechat/EWeChatSender.java [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+        if (Boolean.TRUE.equals(enableProxy)) {
+            httpClient = getProxyClient(proxy, port, user, passwd);
+            RequestConfig rcf = getProxyConfig(proxy, port);
+            httpPost.setConfig(rcf);
+        } else {
+            httpClient = getDefaultClient();
+        }
+
+        try {
+            CloseableHttpResponse response = httpClient.execute(httpPost);
```
Comment (kezhenxu94) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-ewechat/src/main/java/org/apache/dolphinscheduler/plugin/alert/ewechat/EWeChatSender.java]: Consider using `try-with-resources` to simplify the codes.

```suggestion
        try (CloseableHttpResponse response = httpClient.execute(httpPost)) {
```

and remove the `finally` block

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-ewechat/src/main/java/org/apache/dolphinscheduler/plugin/alert/ewechat/exception/EWeChatAlertException.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.plugin.alert.ewechat.exception;
+
+public class EWeChatAlertException extends RuntimeException {
```
Comment (kezhenxu94) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-ewechat/src/main/java/org/apache/dolphinscheduler/plugin/alert/ewechat/exception/EWeChatAlertException.java]: Consider just use `RuntimeException`, providing a separate exception makes no sense in this plugin as nobody will catch this specific exception and do recover

**Review by kezhenxu94 — COMMENTED** [review]
Generally good to me, I left some comments @jon-qj

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-ewechat/src/main/java/org/apache/dolphinscheduler/plugin/alert/ewechat/EWeChatSender.java (hunk 3 above)
Comment (jon-qj) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-ewechat/src/main/java/org/apache/dolphinscheduler/plugin/alert/ewechat/EWeChatSender.java]: Thx.

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-ewechat/src/main/java/org/apache/dolphinscheduler/plugin/alert/ewechat/exception/EWeChatAlertException.java (hunk 4 above)
Comment (jon-qj) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-ewechat/src/main/java/org/apache/dolphinscheduler/plugin/alert/ewechat/exception/EWeChatAlertException.java]: Thx.

**Pull request closed without merging**